src/components/container/MealEditor/MealEditor.tsx

```diff
@@ -1,150 +1,256 @@
-import React, { ChangeEvent, useContext, useState } from 'react';
+import React, { CSSProperties, useContext, useEffect, useState } from 'react';
 import './MealEditor.css';
-import { getMeals, getMealToEdit, getMealTypeDisplayText, language, Meal, saveMeals, timestampSortAsc, useInitializeView } from '../../../helpers';
+import { formatDateForLocale, getMealImageUrls, getMeals, getMealToEdit, getMealTypeDisplayText, language, Meal, resolveColor, saveMeals, timestampSortAsc, uploadMealImageFile } from '../../../helpers';
 import { Button, LayoutContext, LoadingIndicator, UnstyledButton } from '../../presentational';
-import { startOfDay, parse } from 'date-fns';
-import { MealEditorPreviewState, previewData } from './MealEditor.previewData';
+import { parse, startOfDay } from 'date-fns';
+import { createPreviewData, MealEditorPreviewState } from './MealEditor.previewData';
 import { FontAwesomeSvgIcon } from 'react-fontawesome-svg-icon';
-import { faTrashCan } from '@fortawesome/free-solid-svg-icons';
-import { formatDateForLocale, capitalizeForLocale } from '../../../helpers/locale';
-
-type EditMode = 'add' | 'edit';
+import { faEdit, faPlus, faTrashCan } from '@fortawesome/free-solid-svg-icons';
 
 export interface MealEditorProps {
-    previewState?: 'loading' | MealEditorPreviewState
+    previewState?: 'loading' | MealEditorPreviewState;
     onError: () => void;
     onDelete: () => void;
     onSave: () => void;
     onCancel: () => void;
+    withImageCapture?: boolean;
+    children?: React.ReactNode;
     innerRef?: React.Ref<HTMLDivElement>;
 }
 
-export default function (props: MealEditorProps) {
+/**
+ * This component can be used to edit meal log entries.
+ */
+export default function MealEditor(props: MealEditorProps) {
     const layoutContext = useContext(LayoutContext);
 
     const [loading, setLoading] = useState<boolean>(true);
+    const [allMeals, setAllMeals] = useState<Meal[]>();
+    const [activeMeals, setActiveMeals] = useState<Meal[]>();
     const [mealToEdit, setMealToEdit] = useState<Meal>();
-    const [meals, setMeals] = useState<Meal[]>();
-    const [mode, setMode] = useState<EditMode>();
+    const [imageUrl, setImageUrl] = useState<string>();
+    const [imageLoading, setImageLoading] = useState<boolean>(true);
+    const [newImageFile, setNewImageFile] = useState<File>();
+    const [imageUploadError, setImageUploadError] = useState<boolean>(false);
 
-    useInitializeView(() => {
+    useEffect(() => {
```

Comment:
I think for the most part, a re-render on `applicationDidBecomeVisible` makes sense for components that just render existing data.

The difference here is that this is an editor component.  If we were to reload the data each time the app came to the foreground, all pending edits would be lost.  So if you add a description, take a picture, adjust the time, and then get a phone call before hitting save, when you return to the editor, it would reload the meal and wipe out those edits.

At least for this use case, the meals aren't getting modified by any background processes anyways (at least not yet), so there shouldn't be anything to reload.

Something we could consider implementing in the future is to reload the meal on `applicationDidBecomeVisible` just to see if anything has changed in the background.  Then we could try to do some sort of merge with the currently editing meal or at least make the user aware that their changes may overwrite some changes that occurred in the background.  That all gets a bit messy, though, so for now I chose to just assume that we don't need to support a graceful simultaneous-editing scenario.  As such, a one time load up front should be sufficient (and correct) for this particular component.